library-valcon-internship/src/pages/HomePage/HomePage.tsx

```diff
@@ -1,40 +1,69 @@
-import { useEffect, useState } from 'react'
+import { useEffect, useRef, useState } from 'react'
 
 import InfiniteScroll from 'react-infinite-scroll-component'
 import { NavLink } from 'react-router-dom'
 
 import addIcon from '../../assets/icons/add-icon.svg'
 import BookList from '../../components/BookList/BookList'
 import Book from '../../models/Book'
+import Where from '../../models/Where'
 import { getBooks } from '../../services/BookService'
 import './HomePage.css'
 
-const HomePage = () => {
+interface HomePageProps {
+  search: string,
+  filter: Where[]
+  sort: string[]
+}
+
+const HomePage = ({ search, filter, sort }: HomePageProps) => {
   const [ pageNumber, setPageNumber ] = useState(1)
```

Comment:
You can have one object for pageable request and store pageNumber, pageLenght etc. there to avoid too many states and managing one by one